src/lib/utils.ts

```diff
@@ -27,3 +28,28 @@ export const getMessagesFromError = (errBody: CustomErrorBody): string[] => {
 
   return errBody.errors.map(err => getAllErrMessages(err)).flat();
 };
+
+// Define general type for useWindowSize hook, which includes width and height
+export interface Size {
+  width: number | undefined;
+  height: number | undefined;
+}
+
+export function useWindowSize(): Size {
+  const [windowSize, setWindowSize] = useState<Size>({
+    width: undefined,
+    height: undefined,
+  });
+  useEffect(() => {
+    function handleResize() {
+      setWindowSize({
+        width: window.innerWidth,
+        height: window.innerHeight,
+      });
+    }
+    window.addEventListener('resize', handleResize);
+    handleResize();
```

Comment:
I took a deeper dive into this and found that with SSR, this might cause issues since window isn't defined on server side. Although the way this hook is written solves this window undefined issue by waiting until after the component is mounted using `useEffect`,  which happens after hydration, this also led me to find another problem with depending on hook like this for rendering the navbar that also occurs on the main website. 

SSR serves a temporary fully-formated HTML document before the running client-side JS and comparing these two through hydration. I think we're serving the desktop navbar in the initial HTML document until this hook mounts, after which we can check window size and then hydrate with the mobile navbar. This mismatch causes the bar to flash from desktop to mobile on page load. 

What are your thoughts on this? Should we try using media queries instead to see if it would solve this SSR hydration mismatch?